plugins/isecl/tools/setup-isecl.sh

```diff
@@ -0,0 +1,380 @@
+    trap rm -f ~/scs.env EXIT
+
+    sed -i "s/^\(SAN_LIST\s*=\s*\).*\$/\1$SYSTEM_SAN/" ~/scs.env
+    sed -i "s/^\(SCS_ADMIN_USERNAME\s*=\s*\).*\$/\1$SCS_ADMIN_USERNAME/" ~/scs.env
+    sed -i "s/^\(SCS_ADMIN_PASSWORD\s*=\s*\).*\$/\1$SCS_ADMIN_PASSWORD/" ~/scs.env
+    sed -i "s/^\(BEARER_TOKEN\s*=\s*\).*\$/\1$ADMIN_TOKEN/" ~/scs.env
+    sed -i "s/^\(CMS_TLS_CERT_SHA384\s*=\s*\).*\$/\1$CMS_TLS_SHA/" ~/scs.env
+    sed -i "s@^\(AAS_API_URL\s*=\s*\).*\$@\1$AAS_URL@" ~/scs.env
+    sed -i "s@^\(CMS_BASE_URL\s*=\s*\).*\$@\1$CMS_URL@" ~/scs.env
+    sed -i "s@^\(INTEL_PROVISIONING_SERVER\s*=\s*\).*\$@\1$INTEL_PROVISIONING_SERVER@" ~/scs.env
+    sed -i "s@^\(INTEL_PROVISIONING_SERVER_API_KEY\s*=\s*\).*\$@\1$API_KEY@" ~/scs.env
+    sed -i "s/^\(SCS_DB_NAME\s*=\s*\).*\$/\1$SCS_DB_NAME/" ~/scs.env
+    sed -i "s/^\(SCS_DB_USERNAME\s*=\s*\).*\$/\1$SCS_DB_USERNAME/" ~/scs.env
+    sed -i "s/^\(SCS_DB_PASSWORD\s*=\s*\).*\$/\1$SCS_DB_PASSWORD/" ~/scs.env
+
+    echo "Installing SGX Caching Service...."
+    (cd $BINARIES_DIR; ./scs-*.bin)
+    scs status > /dev/null
+    if [ $? -ne 0 ]; then
+	    echo "SGX Caching Service Installation Failed!!"
+	    exit 1
+    fi
+    echo "Installed SGX Caching Service."
+}
+
+install_sqvs()
+{
+    if [ -z "$ADMIN_TOKEN" ];then
+        echo "Empty admin token!!"
+        exit 1
+    fi
+    cp $BINARIES_DIR/env/sqvs.env ~/sqvs.env
+    trap rm -f ~/sqvs.env EXIT
+
+    echo "Updating SGX Quote Verification Service env...."
+    sed -i "s/^\(SAN_LIST\s*=\s*\).*\$/\1$SYSTEM_SAN/" ~/sqvs.env
+    sed -i "s/^\(BEARER_TOKEN\s*=\s*\).*\$/\1$ADMIN_TOKEN/" ~/sqvs.env
+    sed -i "s/^\(CMS_TLS_CERT_SHA384\s*=\s*\).*\$/\1$CMS_TLS_SHA/" ~/sqvs.env
+    sed -i "s@^\(AAS_API_URL\s*=\s*\).*\$@\1$AAS_URL@" ~/sqvs.env
+    sed -i "s@^\(CMS_BASE_URL\s*=\s*\).*\$@\1$CMS_URL@" ~/sqvs.env
+    sed -i "s@^\(SCS_BASE_URL\s*=\s*\).*\$@\1$SCS_URL@" ~/sqvs.env
+    sed -i "s@^\(SGX_TRUSTED_ROOT_CA_PATH\s*=\s*\).*\$@\1$BINARIES_DIR/trusted_rootca.pem@" ~/sqvs.env
+    echo "Installing SGX Quote Verification Service...."
+    (cd $BINARIES_DIR; ./sqvs-*.bin)
+    sqvs status > /dev/null
+    if [ $? -ne 0 ]; then
+    	echo "SGX Quote Verification Service Installation Failed!!"
+    	exit 1
+    fi
+    echo "Installed SGX Quote Verification Service."
+}
+
+install_kbs()
+{
+    if [ -z "$ADMIN_TOKEN" ]; then
+        echo "Empty admin token!!"
+        exit 1
+    fi
+    KBS_HOSTNAME="$(hostname)"
+
+    cp $BINARIES_DIR/env/kbs.env ~/kbs.env
+    trap rm -f ~/kbs.env EXIT
+
+    echo "Updating Key Broker Service env...."
+    sed -i "s/^\(TLS_SAN_LIST\s*=\s*\).*\$/\1$SYSTEM_SAN,$KBS_HOSTNAME/" ~/kbs.env
+    sed -i "s/^\(KBS_SERVICE_USERNAME\s*=\s*\).*\$/\1$KBS_SERVICE_USERNAME/" ~/kbs.env
+    sed -i "s/^\(KBS_SERVICE_PASSWORD\s*=\s*\).*\$/\1$KBS_SERVICE_PASSWORD/" ~/kbs.env
+    sed -i "s/^\(BEARER_TOKEN\s*=\s*\).*\$/\1$ADMIN_TOKEN/" ~/kbs.env
+    sed -i "s/^\(CMS_TLS_CERT_SHA384\s*=\s*\).*\$/\1$CMS_TLS_SHA/" ~/kbs.env
```

Comment:
I'm thinking we could just regenerate the whole env file instead of using sed for every line?

Reply:
> I'm thinking we could just regenerate the whole env file instead of using sed for every line?

Makes sense :-) I need to check that way if we are not missing any pre-existing environment in the file.